pkg/client/userd/service.go

```diff
@@ -115,9 +116,20 @@ func Command(getCommands CommandFactory, daemonServices []DaemonService, session
 	return c
 }
 
+func (s *Service) applySessionConfig(c context.Context) error {
+	cfg, err := client.LoadConfig(c)
+	if err != nil {
+		return err
+	}
+	client.MergeAndReplace(c, s.sessionConfig, cfg)
+	return logging.ReloadDaemonConfig(c, false)
+}
+
 func (s *Service) configReload(c context.Context) error {
-	return client.Watch(c, func(c context.Context) error {
-		return logging.ReloadDaemonConfig(c, false)
+	return client.Watch(c, func(ctx context.Context) error {
+		s.sessionLock.RLock()
+		defer s.sessionLock.RUnlock()
+		return s.applySessionConfig(c)
```

Comment:
Suggestion:
1. Remove the `sessionConfig` field from the Service. As the name suggests, that config is always bound to a session.
2. move the `applySessionConfig()` function to the session and rename it to `applyConfig()`
3. Here you can do `if s.session != nil { s.session.applyConfig(c) }`.